geoips/commandline/geoips_command.py

```diff
@@ -52,6 +52,58 @@ def __init__(self):
 plugin_packages = PluginPackages()
 
 
+class ParentParsers:
+    """Object containing shared arguments for commands in a hierarchical order.
+
+    The name of the parsers in this class directly link to the name of the top-level
+    command which will share its arguments. For example, since our top-level command is
+    'geoips' (see GeoipsCLI), then the arguments created by 'geoips_parser' will be
+    shared amongst all of its child commands. As such, every argument created by
+    'list_parser' will be shared by 'list' commands, and so on. So long as the name of
+    the class matches the structure <class_name>_parser, then the arguments will be
+    shared correctly.
+    """
+
+    geoips_parser = argparse.ArgumentParser(add_help=False)
+    geoips_parser.add_argument(
+        "--log_level",
+        "-log",
+        type=str,
+        default="interactive",
+        choices=["debug", "error", "info", "interactive", "warning"],
+        help="The logging level to use for output via the CLI.",
+    )
+
+    list_parser = argparse.ArgumentParser(add_help=False)
+    list_parser.add_argument(
+        "--package_name",
+        "-p",
+        type=str,
+        default="all",
+        choices=plugin_packages.entrypoints,
+        help="The GeoIPS package to list from.",
+    )
+    mutex_group = list_parser.add_mutually_exclusive_group()
+    mutex_group.add_argument(
+        "--long",
+        "-l",
+        default=True,
+        action="store_true",
+        help="Flag representing the 'long' listing of a certain command.",
+    )
+    mutex_group.add_argument(
+        "--columns",
+        "-c",
+        type=str,
+        nargs="+",
+        default=None,
+        help="""Specific Headers of Data you'd like to see listed.
+                For more in formation on headers available, run
+                'geoips list <cmd> <positional_args> --columns help'.
+                """,
+    )
```

Comment:
I haven't really seen someone use a class like this before. It's interesting! I think this is good for now, but I think we should consider two things:

1. Is there a better container for this than a class?
2. Is there a way to generalize this? Currently this will work only for the top-level commands and it would be difficult to apply propagating arguments to lower-level commands.

I'm running out of time today, but we should write an issue for this.

Reply:
See #715.